File: server/internal/types/url_analysis.go
package types

type BaseURLAnalysisData struct {
	URL           string `gorm:"not null;index" json:"url"`
	HTMLVersion   string `json:"htmlVersion"`
	PageTitle     string `json:"pageTitle"`
	InternalLinks int    `json:"internalLinks"`
	ExternalLinks int    `json:"externalLinks"`
	BrokenLinks   int    `json:"brokenLinks"`
	HasLoginForm  bool   `json:"hasLoginForm"`
	Error         string `json:"error,omitempty"`
}

File: client/src/types/url-analysis.ts
export type URLStatus = "queued" | "running" | "done" | "error" | "stopped"

export interface BrokenLink {
  url: string
  statusCode: number
  error?: string
}

export interface URLEntry {
  id: string
  url: string
  title: string
  htmlVersion: string
  internalLinks: number
  externalLinks: number
  brokenLinks: number
  hasLoginForm: boolean
  headingCounts: Record<string, number>
  brokenLinkDetails: BrokenLink[]
  status: URLStatus
  lastUpdated: Date | string
  processingTime?: number
  crawlProgress?: number
  errorMessage?: string
}
